libnavigator/src/main/java/com/mapbox/navigation/navigator/internal/RouterMapper.kt

```diff
@@ -24,3 +36,26 @@ fun String.mapToRoutingMode(): RoutingMode {
         else -> throw IllegalArgumentException("Invalid routing profile: $this")
     }
 }
+
+/**
+ * https://github.com/mapbox/mapbox-navigation-native/issues/4296
+ * Nav native requires a DirectionsResponse so we are drafting a fake one.
+ * In order to preserve the original request, will require api changes.
+ */
+internal fun List<DirectionsRoute>.mapToDirectionsResponse(
+    routeOptions: RouteOptions
+): DirectionsResponse =
+    DirectionsResponse.builder()
+        .routes(this.toMutableList())
+        .code(HTTP_SUCCESS_CODE)
+        .waypoints(
+            routeOptions.coordinatesList().mapIndexed { index, point ->
+                val waypointBuilder = DirectionsWaypoint.builder()
+                    .rawLocation(doubleArrayOf(point.longitude(), point.latitude()))
+                routeOptions.waypointNamesList()?.getOrNull(index).let { name ->
+                    waypointBuilder.name(name ?: "")
+                }
+                waypointBuilder.build()
+            }
+        )
+        .build()
```

Comment:
Can it be replaced by `DirectionsRoute.toNavigationRoute()` ?

Reply:
it may, but we don't need the whole route there, `DirectionsResponse` only